examples/edge-stack-iso/custom/files/rancher.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+set -euo pipefail
+
+BASEDIR="$(dirname "$0")"
+source ${BASEDIR}/basic-setup.sh
+
+RANCHERLOCKNAMESPACE="default"
+RANCHERLOCKCMNAME="rancher-lock"
+
+if [ -z "${RANCHER_FINALPASSWORD}" ]; then
+  # If there is no final password, then finish the setup right away
+  exit 0
+fi
+
+# Get or create the lock to run all those steps just in a single node
+# As the first node is created WAY before the others, this should be enough
+# TODO: Investigate if leases is better
+if [ $(${KUBECTL} get cm -n ${RANCHERLOCKNAMESPACE} ${RANCHERLOCKCMNAME} -o name | wc -l) -lt 1 ]; then
+  ${KUBECTL} create configmap ${RANCHERLOCKCMNAME} -n ${RANCHERLOCKNAMESPACE} --from-literal foo=bar
+else
+  exit 0
+fi
```

Comment:
The point of this 'lock' is because this script will be executed on all the hosts, so it is a poor's man 'distributed/shared lock' :D If a file is being written by one of the hosts, the others won't be able to see it (unless I'm missing something)

Reply:
This is precisely the case. Isn't Rancher supposed to be bootstrapped just once?

Reply:
I'm sorry but I feel I'm missing something. The script will be executed on all the hosts being provisioned as there is no way to make a distinction right now on EIB. If there is no lock it can happen that all the hosts are waiting for rancher to be up (next line) and then they try to create the password, etc. at the same time and then the script may fail on those that are not the first one.
If I create a file locally, it will be created on each host, leading to the previous scenario as well as there is no 'sync' between them in that case.